### ISSUE TITLE
Request joystick plug auto-connection

### ISSUE THREAD

**Issue by cgutman** [maxiberta/moonlight-snap#2]
Per @popey 's suggestion on https://github.com/libretro/retroarch-snap/issues/11 we should also request this be autoconnected by default to provide an equivalent user experience to the official Flatpak package.

Once this is done and snapd 2.37 makes it out of bionic-proposed to fix #1 , I will add a link to the website to direct Ubuntu users to this Snap.

@maxiberta as package owner, can you create the forum thread required?

**Comment by maxiberta** [comment]
Good idea, thanks! Just requested the auto-connection of the `joystick` interface via [this forum topic](https://forum.snapcraft.io/t/joystick-auto-connection-request-for-moonlight/9532).

It needs 3 +1 votes in one week, as per [the process for aliases, auto-connections and tracks](https://forum.snapcraft.io/t/process-for-aliases-auto-connections-and-tracks/455).

**Comment by maxiberta** [comment]
@cgutman auto-connection of the joystick interface granted and already live :-)

**Comment by popey** [comment]
"Once this is done and snapd 2.37 makes it out of bionic-proposed to fix #1 , I will add a link to the website to direct Ubuntu users to this Snap."

Just FYI, but snaps work on around [50 distros](https://snapcraft.io/core) (see bottom right graph), not just Ubuntu :)

**Comment by cgutman** [comment]
@popey yep, I know. I'm going to be conservative until the Snap gets some user testing. I've learned from shipping the Flatpak version that there are no shortage of strange/ancient system configurations and driver bugs across the diverse Linux user base.

We are exercising APIs that few applications use, like VDPAU and VA-API (hence the VDPAU bug you found). We're also doing it in a way that makes it very obvious when VDPAU or VA-API is broken (the warning dialog), where something like VLC would just fall back to CPU decoding and the user probably never know.

There are benefits and drawbacks to both, so it is nice to have both options to offer users.

The Flatpak has:
- Newer versions of Qt and SDL that match what we use on other platforms
- More testing by users and teething issues already worked out

The Snap has:
- Better experience with Ubuntu and derivatives by being included out of the box with auto-updates
- Significantly smaller download size
- Public usage stats

**Comment by maxiberta** [comment]
@cgutman FTR, the snap *could* use newer/arbitrary versions of Qt and SDL, but they'd need to be compiled from scratch on each snap build (as we're already doing with `ffmpeg`), which sounds non trivial, especially for Qt. But it's something we can explore if really needed (e.g. a hard dependency on a new feature in latest Qt).

**Comment by cgutman** [comment]
Yep, I don't think it's worth it to build Qt right now. There's one bug related to focus handling when opening a context menu using a gamepad that affects the Snap with Qt 5.9 but not the Flatpak with Qt 5.12, but I would rather just find a workaround for that rather than bloat the build and download with a custom Qt build.
